### PR TITLE
Ensure all IServiceCollection extensions are initialized in an idiomatic way.

### LINKED ISSUE #1086: Google.Cloud.Diagnostics.AspNetCore IServiceCollection extension methods are not idiomatic.
Nearly every example of an **Add** extension method to `IServiceCollection` that takes arguments takes them by taking a setup `Action` and sending it some sort of options object.
```
            services.AddMvc(options =>
            {
                options.RespectBrowserAcceptHeader = true;
            });
```

Examples include AddAntiForgery, AddAuthorization, AddLocalization and many others.

### PR DESCRIPTION
**Pull request by iantalarico**
Fixes #1086

### CODE REVIEW

File: apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/ErrorReporting/ErrorReportingServiceOptions.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    public sealed class ErrorReportingServiceOptions
+    {
+        /// <summary>
+        /// The Google Cloud Platform project ID. If unspecified and running on GAE or GCE the project
+        /// ID will be detected from the platform.
+        /// </summary>
+        public string ProjectId { get; set; }
+
+        /// <summary>
+        /// An identifier of the service, such as the name of the  executable or job. Cannot be null.
```
Comment (evildour) [inline apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/ErrorReporting/ErrorReportingServiceOptions.cs]: remove extra space in `...the  executable...`

File: apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/ErrorReporting/ErrorReportingExceptionLoggerExtension.cs [hunk 2]
```diff
@@ -98,33 +78,21 @@ public static void UseGoogleExceptionLogging(this IApplicationBuilder app)
...
-        ///     Cannot be null.</param>
-        /// <param name="options">Optional, error reporting options.</param>
-        private static void AddGoogleExceptionLoggingBase(
-            this IServiceCollection services, string projectId, string serviceName, string version,
-            ErrorReportingOptions options = null)
-        {
             services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();
-            services.AddSingleton(ErrorReportingContextExceptionLogger.Create(projectId, serviceName, version, options));
+            services.AddSingleton(ErrorReportingContextExceptionLogger.Create(
+                serviceOptions.ProjectId, serviceName, version, serviceOptions.Options));
```
Comment (evildour) [inline apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/ErrorReporting/ErrorReportingExceptionLoggerExtension.cs]: Should you use `CommonUtils.GetAndCheckProjectId` here?

File: apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/ErrorReporting/ErrorReportingServiceOptions.cs (hunk 1 above)
Comment (iantalarico) [inline apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/ErrorReporting/ErrorReportingServiceOptions.cs]: Done

File: apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/ErrorReporting/ErrorReportingExceptionLoggerExtension.cs (hunk 2 above)
Comment (iantalarico) [inline apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/ErrorReporting/ErrorReportingExceptionLoggerExtension.cs]: No, it's called further down in the stack as it's not required that a PID is used.   A user could log to an organization (this is not common as far as I've seen).

The PID option is a convince as this is how most users will probably log.

**Pull request merged**